cpp/collections_module/algorithms/collections.hpp

```diff
@@ -0,0 +1,29 @@
+#include <mgp.hpp>
+
+constexpr std::string_view kReturnValueUnionAll = "return_list";
+constexpr std::string_view kProcedureUnionAll = "unionAll";
+constexpr std::string_view kArgumentList1UnionAll = "list1";
+constexpr std::string_view kArgumentList2UnionAll = "list2";
+
+
+
+void unionAll(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory){
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try{
+    mgp::List list1 = arguments[0].ValueList();
+    mgp::List list2 = arguments[1].ValueList();
+    
+    for (size_t i = 0; i < list2.Size(); i++){
+        list1.AppendExtend(std::move(list2[i]));
+    }
+    auto record = record_factory.NewRecord();
+    record.Insert(std::string(kReturnValueUnionAll).c_str(), list1);
```

Comment:
I have an idea here to extend CPP API so we can move the list here instead of doing copy underlying. Ping me on Monday so we can sync what to do here.